### PR TITLE
Detect pull request authors for issues

### PR DESCRIPTION
**Pull request by andrewda**
This detects the authors of the pull requests that close issues in
order to link them with the student who claimed the task.

Finds 3 unlinked users: @nalinbhardwaj, @ravernkoh and @danes44 

Closes https://github.com/coala/community/issues/56

### CODE REVIEW

**Comment by nkprince007** [comment]
Hey, @andrewda, an MR with IGitt has `get_closing_issues` which can retrieve the issues it can close.

**Comment by andrewda** [comment]
@nkprince007 In this case we want to go the opposite direction, though, right? We have the link to an issue and we want to retrieve which PR/MR closes it. It'd be cool if this were in GitHub's API (sadly it's not), but it could also be part of IGitt, I guess.

**Comment by nkprince007** [comment]
@andrewda, yeah, it'd be really appreciated, if you'd do a PR to IGitt as well ;)

File: gci/students.py [hunk 1]
```diff
@@ -129,6 +130,39 @@ def get_issue_related_students():
...
+def get_pr_closing_issue(url):
+    r = requests.get(url)
+    result = re.search(
+        'closed this(?:\n| )+in(?:\n| )+<a href=\"/(.+)/pull/(?:[0-9]+)\">#' +
+        '([0-9]+)</a>',
+        r.text
+    )
+
+    if result is not None:
+        return {
```
Comment (jayvdb) [inline gci/students.py]: This should return a `MergeRequest` , which has a `.author` added in https://gitlab.com/gitmate/open-source/IGitt/merge_requests/159 ?

But also, this function should be in `IGitt`.

File: gci/students.py [hunk 2]
```diff
@@ -160,9 +194,18 @@ def get_linked_students():
...
+                        user = get_closing_pr_author(url)
+                        if user is not None:
+                            student['username'] = user
+                            print('student %s is %s because of %s (from PR)' %
+                                  (student['id'], user, url))
+                            yield student
+                            break
                     else:
-                        student['username'] = issue.assignees[0]
+                        student['username'] = list(issue.assignees)[0]
```
Comment (jayvdb) [inline gci/students.py]: this seems to be unrelated to your issue.  I assume it is caused by the updated IGitt version, in which case you need a separate commit for the IGitt update and these fixes

File: gci/students.py (hunk 1 above)
Comment (jayvdb) [reply gci/students.py]: Also note that issues can be re-opened, and have a second commit which closes them again, so there is no 1-1 relationship.

**Comment by andrewda** [comment]
https://gitlab.com/gitmate/open-source/IGitt/merge_requests/197

**Comment by jayvdb** [comment]
ack https://github.com/coala/community/pull/69/commits/ed427bc7d5bcc1bc715dcf2fe23c3c5a385a9031 https://github.com/coala/community/pull/69/commits/105a0b7bdc35558d146395ddac9aa55ec7fd6b8b

**Comment by jayvdb** [comment]
@gitmate-bot ff

File: gci/students.py [hunk 3]
```diff
@@ -161,8 +161,24 @@ def get_linked_students():
...
                         print('student %s is %s because of %s' %
-                              (student['id'], issue.assignees[0], url))
+                              (student['id'], student['username'], url))
+                        yield student
+                        break
+
+                    if len(issue.mrs_closed_by) == 0:
+                        logger.info('task %d: No mrs closing %s' %
+                                    (task_id, url))
+                    elif len(issue.mrs_closed_by) == 0:
```
Comment (nalinbhardwaj) [inline gci/students.py]: Shouldn’t this be `len(issue.mrs_closed_by) > 1`
